src/mon/OSDMonitor.cc

```diff
@@ -7703,6 +7703,13 @@ bool OSDMonitor::preprocess_pool_op(MonOpRequestRef op)
 {
   op->mark_osdmon_event(__func__);
   MPoolOp *m = static_cast<MPoolOp*>(op->get_req());
+  
+  if (m->fsid != mon->monmap->fsid) {
+    dout(0) << "preprocess_pool_op: on fsid " << m->fsid
+            << " != " << mon->monmap->fsid << dendl;
+    return true;
```

Comment:
passing an error string via the `MPoolOpReply ::response_data` is a viable way, but we need to change the interface of librados to return the error string. so returning an error code is fine here in the updated commit.